src/main/scala/com/softwaremill/realworld/articles/ArticlesService.scala

```diff
@@ -1,38 +1,27 @@
 package com.softwaremill.realworld.articles
 
-import com.softwaremill.realworld.profiles.ProfilesRepository
+import com.softwaremill.realworld.profiles.ProfileRow
+import com.softwaremill.realworld.utils.Exceptions
 import zio.{IO, ZIO, ZLayer}
 
-class ArticlesService(articlesRepository: ArticlesRepository, profilesRepository: ProfilesRepository):
+import java.sql.SQLException
+import javax.sql.DataSource
 
-  def list(): IO[String, List[Article]] = articlesRepository
+class ArticlesService(articlesRepository: ArticlesRepository):
+
+  def list(): ZIO[DataSource, Exception, List[Article]] = articlesRepository
     .list()
-    .flatMap(sas => ZIO.collectAll(sas.map(article)))
     .onError(err => ZIO.logError(err.prettyPrint))
 
-  private def article(a: StoredArticle): IO[String, Article] = {
-    profilesRepository
-      .find(a.authorId)
-      // TODO implement "following"
-      .map(p => ArticleAuthor(p.username, p.bio, p.image, following = false))
-      .map(author =>
-        Article(
-          a.slug,
-          a.title,
-          a.description,
-          a.body,
-          a.tagList,
-          a.createdAt,
-          a.updatedAt,
-          // TODO implement "favorited", "favoritesCount"
-          favorited = false,
-          favoritesCount = 0,
-          author
-        )
-      )
-      .mapError(_ => s"Author (id: ${a.authorId} not found for article '${a.slug}'.")
-  }
+  def find(slug: String): ZIO[DataSource, Exception, Article] = articlesRepository
+    .find(slug)
+    .foldZIO(
```

Comment:
isn't this a `.map`?

Reply:
Repository returns `Option[Article]` and I want to change `None` to the exception. I think I can't do this with map.

Reply:
Tweaked, along the changes to error handling.